app/src/main/java/org/fossasia/openevent/general/auth/EditProfileFragment.kt

```diff
@@ -53,14 +55,25 @@ class EditProfileFragment : Fragment() {
     ): View? {
         rootView = inflater.inflate(R.layout.fragment_edit_profile, container, false)
 
+        DataBindingUtil.setContentView<FragmentEditProfileBinding>(
+            activity as Activity, R.layout.fragment_edit_profile
+        ).apply {
+            this.setLifecycleOwner(activity)
+            this.viewmodel = editProfileViewModel
+        }
+
         profileViewModel.user
             .nonNull()
             .observe(this, Observer {
                 val userFirstName = it.firstName.nullToEmpty()
                 val userLastName = it.lastName.nullToEmpty()
                 val imageUrl = it.avatarUrl.nullToEmpty()
-                rootView.firstName.setText(userFirstName)
-                rootView.lastName.setText(userLastName)
+                if (editProfileViewModel.firstName.value.isNullOrEmpty()) {
+                    editProfileViewModel.firstName.value = userFirstName
+                }
+                if (editProfileViewModel.lastName.value.isNullOrEmpty()) {
+                    editProfileViewModel.lastName.value = userLastName
+                }
```

Comment:
What to do to correct it? As this method is called everytime the fragment is created. @iamareebjamal

Reply:
Am using this to store the names for the first time and added if else so that the MutableLiveData does not get the same value again on rotation. @iamareebjamal

Reply:
Shouldn't be needed at all

Reply:
I tried setting the data to mutable live data directly but its not working. @iamareebjamal

Reply:
If I remove if-else checks then on rotation the initial first and last names are set. Live changes are not reflected.

Reply:
If I remove if else then the initial first names and last names are set on rotation to the mutableLiveData @iamareebjamal. Live changes are not reflected